provider/status_changes.json

```diff
@@ -300,6 +300,28 @@
                 }
               }
             ],
+            "anyOf": [
```

Comment:
Is this just `anyOf` instead of `oneOf` because you can't specify `oneOf` twice for the same object? Not to get too far into the weeds, but wouldn't it be more semantically correct to structure this whole sub-schema as a single `allOf` with three sub-schemas: 1. the current `oneOf` defined above for valid `event_type` and `event_type_reason` combinations
2. this, as a `oneOf`, defining when `associated trips` is required
3. the rest of the properties, defined below.

More specifically, this would look like (non-relevant intermediate lines omitted for brevity):
```json
"items": {
  "$id": ...,
  ...
  "required": [...],
  "allOf": [
    { "oneOf": [ "schema_for_valid_event_types_and_reasons" ] },
    { "oneOf": [ "this_schema_for_associated_trip" ] },
    { "properties": "...properties listed below...." }
  ]
}
```

(caveat this is the first time I've read the JSON schema spec closely, so take this with a dose of skepticism/it may be way wrong)

Reply:
tagging @ian-r-rose,